vividus-facade-jira/src/main/java/org/vividus/jira/JiraClient.java

```diff
@@ -16,35 +16,55 @@
 
 package org.vividus.jira;
 
+import static org.apache.commons.lang3.Validate.isTrue;
+
 import java.io.IOException;
+import java.util.function.Supplier;
 
 import org.apache.http.HttpEntity;
 import org.apache.http.HttpStatus;
 import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.client.protocol.HttpClientContext;
 import org.apache.http.entity.ContentType;
 import org.apache.http.entity.StringEntity;
 import org.apache.http.protocol.HttpContext;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.vividus.http.HttpMethod;
 import org.vividus.http.HttpRequestBuilder;
+import org.vividus.http.client.ClientBuilderUtils;
 import org.vividus.http.client.HttpResponse;
 import org.vividus.http.client.IHttpClient;
-import org.vividus.http.context.HttpContextFactory;
 
 public class JiraClient
 {
     private static final Logger LOGGER = LoggerFactory.getLogger(JiraClient.class);
 
     private final String endpoint;
     private final IHttpClient httpClient;
-    private final HttpContextFactory httpContextFactory;
+    private final Supplier<HttpContext> httpContextFactory;
 
-    public JiraClient(String endpoint, IHttpClient httpClient, HttpContextFactory httpContextFactory)
+    public JiraClient(String endpoint, String username, String password, IHttpClient httpClient)
     {
+        if (username != null && password != null)
+        {
+            this.httpContextFactory = () ->
+            {
+                HttpContext httpContext = HttpClientContext.create();
+                httpContext.setAttribute(HttpClientContext.CREDS_PROVIDER,
+                        ClientBuilderUtils.createCredentialsProvider(username, password));
+                return httpContext;
+            };
+        }
+        else
+        {
+            isTrue(username == null && password == null, "The JIRA %s is missing",
+                    username == null ? "username" : "password");
+            httpContextFactory = () -> null;
```

Comment:
either use `this` in both cases or nowhere

Reply:
:((((((